packages/app/cypress/e2e/integration/top-nav.spec.ts

```diff
@@ -289,31 +289,118 @@ describe('App Top Nav Workflows', () => {
+        mockLogInActionsForUser(mockUser)
+
+        cy.findByTestId('app-header-bar').within(() => {
+          cy.findByTestId('user-avatar').should('not.exist')
+          cy.findByRole('button', { name: 'Log In' }).click()
+        })
+
+        cy.findByRole('dialog', { name: 'Log in to Cypress' }).as('logInModal').within(() => {
+          cy.findByRole('button', { name: 'Log In' }).click()
 
-        cy.findByTestId('app-header-bar').findByRole('button', { name: 'Log In' }).as('logInButton')
+          // The Log In button transitions through a few states as the browser launch lifecycle completes
+          cy.findByRole('button', { name: 'Opening Browser' }).should('be.visible').and('be.disabled')
+          cy.findByRole('button', { name: 'Waiting for you to log in' }).should('be.visible').and('be.disabled')
+        })
+
+        cy.findByRole('dialog', { name: 'Login Successful' }).within(() => {
+          cy.findByText('You are now logged in as', { exact: false }).should('be.visible')
+          cy.validateExternalLink({ name: mockUser.name, href: 'https://on.cypress.io/dashboard/profile' })
+
+          // The dialog can be closed at this point by either the header close button or the Continue button
+          // The Continue button is tested here
+          cy.findByRole('button', { name: 'Close' }).should('be.visible').and('not.be.disabled')
+          cy.findByRole('button', { name: 'Continue' }).click()
+        })
+
+        cy.get('@logInModal').should('not.exist')
+        cy.findByTestId('app-header-bar').findByTestId('user-avatar').should('be.visible')
       })
 
-      it('shows log in modal when button is pressed', () => {
-        cy.get('@logInButton').click()
+      it('shows log in modal workflow for user with only email', () => {
+        mockLogInActionsForUser(mockUserNoName)
+
+        cy.findByTestId('app-header-bar').within(() => {
+          cy.findByTestId('user-avatar').should('not.exist')
+          cy.findByRole('button', { name: 'Log In' }).click()
+        })
+
+        cy.findByRole('dialog', { name: 'Log in to Cypress' }).as('logInModal').within(() => {
+          cy.findByRole('button', { name: 'Log In' }).click()
+
+          // The Log In button transitions through a few states as the browser launch lifecycle completes
+          cy.findByRole('button', { name: 'Opening Browser' }).should('be.visible').and('be.disabled')
+          cy.findByRole('button', { name: 'Waiting for you to log in' }).should('be.visible').and('be.disabled')
+        })
+
+        cy.findByRole('dialog', { name: 'Login Successful' }).within(() => {
+          cy.findByText('You are now logged in as', { exact: false }).should('be.visible')
+          cy.validateExternalLink({ name: mockUserNoName.email, href: 'https://on.cypress.io/dashboard/profile' })
+
+          // The dialog can be closed at this point by either the header close button or the Continue button
```

Comment:
Can the dialog be closed by outside click or by pressing the escape key? Do we need to validate the keyboard interactions with the close button or is this handled by component tests?

Reply:
Outside click, esc key, keyboard interaction with button are all valid ways to back out of this modal. There's a component test coming soon that should cover all of that at the level of the modal component that is used for every modal. It's no harm to have some coverage in E2E somewhere too though.

The outside-click behavior is configurable - for example, some modals are not closed when you click away, and that's intentional. So for any given modal we might as well assert that it closes when you click outside or not, since in theory there's a logical reason why some are less easily dismissed.